api/v1alpha1/nicclusterpolicy_webhook.go

```diff
@@ -347,6 +375,12 @@ func isValidRdmaSharedDevicePluginResourceName(resourceName string) bool {
 	return resourceNameRegex.MatchString(resourceName)
 }
 
+func isValidFQDN(input string) bool {
+	pattern := `^[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z]{2,})+$`
```

Comment:
could you put this one into a const ? 
then use it here and in the err msg when checking resource name ?

you can try to use raw string for it: https://www.golangprograms.com/golang-raw-string-literals-and-interpreted-string-literals.html

if it doesnt work, im ok with keeping it as is.

also while at it,  would be great if you did the same for the resource name regex as well.

Reply:
Done